### ISSUE TITLE
Type signature

### ISSUE THREAD

**Issue by riton** [ccin2p3/puppet-etc_services#3]
Thanks for releasing this module on Puppet Forge. I'm using it for my own company, but I've made some changes to how the type is specified. Instead of:

```
    etc_services { 'kerberos/udp':
      port    => '88',
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
    }
```

I use:

```
    etc_services { 'kerberos':
      ports    => ['tcp/88',
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
    }
```

This more closely matches the /etc/services syntax.

**Comment by riton** [comment]
@alexjurkiewicz submitted this thread initially.

**Comment by faxm0dem** [comment]
:+1: at this point I feel it is not necessary to keep backwards compatibility

**Comment by alexjurkiewicz** [comment]
Actually, there's a problem. You can't specify tcp & udp with the same $name. You could either keep the current design or have an ugly solution (since I think it's rare to have this situation):

    etc_services { 'kerberos udp':
        title => 'kerberos',
        port => 'udp/88',
     }
    etc_services { 'kerberos tcp':
        title => 'kerberos',
        port => 'tcp/88',
     }
    etc_services { 'http':
        port => 'tcp/80',
     }

Personally, I don't need tcp & udp entries, so I'm happy to have an ugly solution for that case. But I'll track this module whatever you decide.

**Comment by riton** [comment]
@alexjurkiewicz I was actually thinking that you introduced a typo in your proposal.

I was thinking that your syntax was:

```
etc_services { 'kerberos':
      ports    => ['tcp/88'], # <==
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
}
```

and could be used to define both _udp_ and _tcp_ entries like:

```
etc_services { 'kerberos':
      ports    => ['tcp/88', 'udp/88'], # <==
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
}
```

What I like about your proposal is the unique `$name` that uniquely identify a service in `/etc/services` (for both _udp_ and _tcp_ entries).

Maybe that the best solution would be to factorize the _port_ information like in:

```
etc_services { 'kerberos':
      protocol => [ 'udp', 'tcp' ],
      port       => 88,
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
}
```

I don't really see __ANY__ use case that could justify a different _udp_ and _tcp_ port for a service.

@faxm0dem @pyfontan @mpuel what do you think about this ?

**Comment by alexjurkiewicz** [comment]
(Y) from me

**Comment by riton** [comment]
After discussion with @faxm0dem , I agree with him that it could be convenient for users to be able to specify a different _tcp_ and _udp_ port.
I wasn't able to find something in rfc6335 that explicitly says that this is impossible.

So I think that

```
etc_services { 'kerberos':
      ports    => ['tcp/88', 'udp/88'],
      aliases => [ 'kerberos5', 'krb5', 'kerberos-sec' ],
      comment => 'Kerberos v5'
}
```

will be adopted.

Thoughts ?

**Comment by riton** [comment]
Fixed by #8
